public/app/plugins/datasource/prometheus/language_provider.ts

```diff
@@ -78,9 +78,16 @@ export default class PromQlLanguageProvider extends LanguageProvider {
   };
 
   // Keep this DOM-free for testing
-  provideCompletionItems({ prefix, wrapperClasses, text }: TypeaheadInput, context?: any): TypeaheadOutput {
+  provideCompletionItems({ prefix, wrapperClasses, text, value }: TypeaheadInput, context?: any): TypeaheadOutput {
```

Comment:
I feel that this function is a bit hard to read. One part is the layout of else-if in javascript, not much to do about that.  But I think breaking out some of the logic to smaller functions with descriptive names could increase the readability.

Reply:
While I agree that it's hard to read, this function is ultimately a dispatcher. I need to prepare some variables on which the if/else chain dispatches. I'm at a loss on what else to do. The breakout took place by splitting extracting the various `get...Items()`. Do you have a concrete suggestion?